pennylane/ops/qubit/parametrized_hamiltonian.py

```diff
@@ -0,0 +1,198 @@
+"""
+This submodule contains the time-dependent hamiltonian class
+"""
+import pennylane as qml
+from pennylane.operation import Observable, Tensor
+from pennylane.ops.qubit.hamiltonian import Hamiltonian
+
+
+# pylint: disable= too-many-instance-attributes
+class ParametrizedHamiltonian(Observable):
+    r"""Operator representing a time-dependent Hamiltonian.
+
+    The Hamiltonian is represented as a linear combination of other operators, e.g.,
+    :math:`H(v, t) = H_\text{drift} + \sum_j f_j(v, t) H_j`, where the :math:`v, t` are trainable parameters,
```

Comment:
```suggestion
    :math:`H(v, t) = H_\text{drift} + \sum_j f_j(v, t) H_j`, where the :math:`v` are trainable parameters,
```

Reply:
Maybe a more specific example could help, e.g., with some sines, cosines, XXs and YYs.

Reply:
Added something. I think the documentation will still need some more work, but will leave as is for now and do that as a separate PR.